tutorials/chapter2.md

```diff
@@ -318,3 +318,81 @@ As you can see from the code, the application we intended (publish "Hello World"
 The ```kpsr::dds_mdlw::DDSEnv``` class is the Environment class built to share and access global data parameters over the DDS distributed environment. The Environment class defines the interface used by the ```kpsr::dds_mdlw::DDSEnv``` to access the parameters. These parameters are loaded from a YAML file into a YamlEnvironment class, and are made available to the DDS environment via a specific pair of DDS Reader/Writers. As a result, these parameters are accessible over the entire DDS environment, and any changes to these parameters are also visible to Klepsydra applications.
 
 ### ZMQ
+
+Klepsydra compiled with ZMQ support generates the `kpsr_zmq_core` library which is available to be linked by applications. In order to send or receive data using ZMQ, Klepsydra provides three ways of serializing data - JSON encoding, binary encoding, or raw data. When dealing with primitive data types, no additional steps are required and we show below how the previous simple example would be used.
+
+#### Hello World example
+
+```cpp
+
+int main (int argv, char ** argc) {
+
+    // Set up zmq specific options.
+    std::string serverUrl = "tcp://*:9001";
+    std::string topic = "example1";
+    zmq::context_t context (1);
+    zmq::socket_t publisher (context, ZMQ_PUB);
+    publisher.bind(serverUrl);
+    std::string clientUrl = "tcp://localhost:9001";
+    zmq::socket_t subscriber (context, ZMQ_SUB);
+    subscriber.connect(clientUrl);
+    subscriber.setsockopt(ZMQ_SUBSCRIBE, topic.c_str(), topic.size());
+
+    // ZMQ set up over, now our simple example:
+
+    kpsr::zmq_mdlw::ToZMQMiddlewareProvider toZMQMiddlewareProvider(nullptr, publisher);
+    kpsr::Publisher<std::string> * toZMQPublisher = toZMQMiddlewareProvider.getJsonToMiddlewareChannel<std::string>(topic, 0);
+    
+    SimplePublisher publisher(toZMQPublisher);
+    
+    //  Process 100 updates
+    kpsr::zmq_mdlw::FromZmqMiddlewareProvider _fromZmqMiddlewareProvider;
+    kpsr::zmq_mdlw::FromZmqChannel<std::string> * _jsonFromZMQProvider = _fromZmqMiddlewareProvider.getJsonFromMiddlewareChannel<std::string>(subscriber, 100);
+    _jsonFromZMQProvider->start();
+    
+    kpsr::high_performance::EventLoopMiddlewareProvider<16> stringDataSafeQueueProvider(nullptr);
+    stringDataSafeQueueProvider.start();
+
+    _jsonFromProvider->registerToTopic(topic, stringDataSafeQueueProvider.getPublisher<std::string>("string", 0, nullptr, nullptr));
+
+    stringDataSafeQueueProvider.getSubscriber<std::string>("string")->registerListener("example", [](std::string &msg) {
+        std::cout << "Received message: " << msg << std::endl; });
```

Comment:
indentation

Reply:
I've checked the source and since this is the lambda function, it is correctly indented with 8 spaces (compared to 4 for the previous line).